### PR TITLE
Modularization III-ish

### PR DESCRIPTION
**Pull request by elsirion**
This is an attempt at finishing modularization in a way that is quick, will allow external modules soon, but leaves most of the code moving of existing modules for later.

### CODE REVIEW

File: fedimint-server/src/net/api.rs [hunk 1]
```diff
@@ -181,8 +139,60 @@ fn attach_endpoints<M>(
...
+        // and path has to live till the end of program anyways.
+        let path: &'static _ =
+            Box::leak(format!("/{}{}", base_name, endpoint.path).into_boxed_str());
+        let handler: &'static _ = Box::leak(endpoint.handler);
+
+        rpc_module
+            .register_async_method(path, move |params, state| {
+                Box::pin(async move {
+                    let params = params.one::<serde_json::Value>()?;
+                    // Using AssertUnwindSafe here is far from ideal. In theory this means we could
```
Comment (dpc) [inline fedimint-server/src/net/api.rs:161]: BTW. the whole `UnwindSafe` is one of a few broken things in Rust and possibly will be deprecated altogether. https://github.com/rust-lang/rust/issues/40628

File: modules/fedimint-ln/src/lib.rs [hunk 2]
```diff
@@ -250,12 +255,17 @@ impl FederationModuleConfigGen for LightningModuleConfigGen {
...
+    type Decoder = LightningModuleDecoder;
+    type Input = LightningInput;
+    type Output = LightningOutput;
+    type OutputOutcome = LightningOutputOutcome;
     type ConsensusItem = LightningConsensusItem;
-    type VerificationCache = ();
+    type VerificationCache = LightningVerificationCache;
+
+    fn module_key(&self) -> fedimint_api::encoding::ModuleKey {
+        todo!()
```
Comment (dpc) [inline modules/fedimint-ln/src/lib.rs:267]: Intentional or leftover?

**Review by dpc — DISMISSED** [review]

File: modules/fedimint-ln/src/lib.rs (hunk 2 above)
Comment (elsirion) [reply modules/fedimint-ln/src/lib.rs:267]: Good catch!

File: fedimint-server/src/net/api.rs (hunk 1 above)
Comment (elsirion) [reply fedimint-server/src/net/api.rs:161]: I think it's a separate issue (created #943), I just copied the old code to adapt it.

**Review by justinmoon — APPROVED** [review]
LGTM

**Pull request merged**